sips/sip-010/sip-010-fungible-token-standard.md

```diff
@@ -0,0 +1,227 @@
+
+This method must be defined with `define-public`, as it alters state, and should be externally callable.
+
+Contract implementers should take note to perform authorization of the `transfer` method. For example, a fungible token contract that wants to make sure that only the transaction's sender is able to move the requested tokens might first check that the sender argument is equal to tx-sender.
+
+When returning an error in this function, the error codes should follow the same patterns as the built-in `ft-transfer?` and `stx-transfer?` functions.
+
+| error code | reason                                          |
+| ---------- | ----------------------------------------------- |
+| u1         | `sender` does not have enough balance           |
+| u2         | `sender` and `recipient` are the same principal |
+| u3         | `amount` is non-positive                        |
+| u4         | `sender` is not the same as `tx-sender`         |
+
+### Name
+
+`(get-name () (response (string-ascii 32) uint))`
+
+Return a human-readable name for the contract, such as "CoolPoints", etc.
+
+This method should be defined as read-only, i.e. `define-read-only`.
+
+### Symbol
+
+`(get-symbol () (response (string-ascii 32) uint))`
+
+Return a symbol that allows for a shorter representation of a token. This is sometimes referred to as a "ticker". Examples: "STX", "COOL", etc. Typically, a token could be referred to as $SYMBOL when referencing it in writing.
+
+This method should be defined as read-only, i.e. `define-read-only`.
+
+### Decimals
+
+`(get-decimals () (response uint uint))`
+
+The number of decimal places in a token. All fungible token balances must be represented as integers, but providing the number of decimals provides for an abstraction of a token that humans are more familiar dealing with. For example, the US Dollar has 2 decimals, if the base unit is "cents", as is typically done in accounting. Stacks has 6 decimals, Bitcoin has 8 decimals, and so on.
+
+As another example, if a token has 4 decimals, and the `get-balance-of` method a particular user returns `100345000`, wallets and exchanges would likely represent that value as `10034.5`.
+
+This method should be defined as read-only, i.e. `define-read-only`.
+
+### Balance of
+
+`(get-balance-of (principal) (response uint uint))`
+
+Return the balance of a particular principal (also known as "address" or "account"). Implementations should typically use the built-in Clarity method `ft-get-balance`.
+
+This method should be defined as read-only, i.e. `define-read-only`.
+
+### Total supply
+
+`(get-total-supply () (response uint uint))`
+
+Return the total supply of this token. Implementations should typically use the built-in Clarity method `ft-get-supply`.
```

Comment:
What to put if total-supply is unlimited?

Reply:
Do you have an example of a token with unlimited liquid supply?

Reply:
the supply may be unlimited, but this is meant to be the current supply if you sum all the addresses that contain the token.  So always a specific value the token should keep track of